Essentials/src/com/earth2me/essentials/perm/PermissionsHandler.java

```diff
@@ -109,35 +109,39 @@ public boolean tryProvider() {
 
     public void checkPermissions() {
         // load and assign a handler
-        List<Class<? extends SuperpermsHandler>> providerClazz = Arrays.asList(
-                LuckPermsHandler.class,
-                ModernVaultHandler.class,
-                GenericVaultHandler.class,
-                SuperpermsHandler.class
-        );
-        for (Class<? extends IPermissionsHandler> providerClass : providerClazz) {
-            try {
-                IPermissionsHandler provider = providerClass.newInstance();
-                if (provider.tryProvider()) {
-                    if (provider.getClass().isInstance(this.handler)) {
-                        return;
+        if (useSuperperms) {
+            List<Class<? extends SuperpermsHandler>> providerClazz = Arrays.asList(
+                    LuckPermsHandler.class,
+                    ModernVaultHandler.class,
+                    GenericVaultHandler.class,
+                    SuperpermsHandler.class
+            );
+            for (Class<? extends IPermissionsHandler> providerClass : providerClazz) {
+                try {
+                    IPermissionsHandler provider = providerClass.newInstance();
+                    if (provider.tryProvider()) {
+                        if (provider.getClass().isInstance(this.handler)) {
+                            return;
+                        }
+                        if (this.handler != null) {
+                            unregisterContexts();
+                        }
+                        this.handler = provider;
+                        initContexts();
+                        break;
                     }
-                    if (this.handler != null) {
-                        unregisterContexts();
-                    }
-                    this.handler = provider;
-                    initContexts();
-                    break;
+                } catch (Throwable ignored) {
                 }
-            } catch (Throwable ignored) {
             }
-        }
-        if (handler == null) {
-            if (useSuperperms) {
+            if (handler == null) {
+                // How?
```

Comment:
https://github.com/EssentialsX/Essentials/blob/e0ec990d2289f30d69a025f15887740ff1140fe5/Essentials/src/com/earth2me/essentials/perm/PermissionsHandler.java#L16
It's initialized to null, and remains null if it fails to load the handler (although unlikely) 😛

Reply:
I guess, just doesn't make sense as the SuperpermsHandler is also in the list of handlers it tries to load.

Reply:
Yeah it'd better load, or Essentials probably has other problems to worry about 😆